caikit/core/data_model/dataobject.py

```diff
@@ -284,6 +318,28 @@ def _to_jtd_schema(input_schema: _SCHEMA_DEF_TYPE) -> _JTD_DEF_TYPE:
         raise
 
 
+def _is_optional_type(type_: Type) -> bool:
+    origin = typing.get_origin(type_)
```

Comment:
NIT: could use a docstring even though it's tiny since this really is where the crux of the optional semantics are defined